CMakeLists.txt

```diff
@@ -44,6 +44,7 @@
 cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
 
 set(CALAMARES_VERSION 3.3.0-alpha1)
+set(CALAMARES_SOVERSION 3.3)
```

Comment:
If you move this below the `project()` command you can use the major, minor parts and don't have to repeat it here. You could even skip setting this entirely and just use major.minor down below, **but** you could also just choose to set this to "4" and call that the soversion.

Reply:
There has never been a promise of ABI stability in Calamares; the functions in *libcalamares* have changed a lot over the course of the 3.2 series, always with the Calamares version as SO version: there's no guarantee any two of the 3.2.x versions were in any way compatible with each other. Once 3.3 is actually released, then we **might** consider ABI stability (in which case 3.3, or 4, might make sense as an SO version).

If we use the `ci/abicheck.sh` script we could keep track of actual compatibility -- otherwise the major.minor SO version is just a lie. You can make use of that script: for instance, run it and add the output to `ci/` as, I dunno, `abi-so-3.3`. Then, in the release script, run the abi check and compare against the stored text. Then we can make sure that the SO version is useful and correct.